index.ts

```diff
@@ -0,0 +1,39 @@
+function fizzbuzz() {
+    for (let i = 1; i < 256; i++) {
+
+        let output: string[] = []
+        let rules = {3: "Fizz", 5: "Buzz", 7: "Bang", 11: "Bong", 13: "Fezz"}
+
+        const checkDivisor = (i: number, divisor: number): boolean => {
+            return i % divisor === 0
+        }
+
+        const applyRule = (i: number, divisor: number) => {
+            if (checkDivisor(i, divisor)) {
```

Comment:
Nit: I'm not sure how useful having a `checkDivisor` is here, I think you could easily just put `i & divisor === 0` here and it would still be readable. I think general rule of thumb I go with is if its self explanatory and on a single line, it doesn't need to be extracted into a separate function. Multiple lines then definitely extract it out